src/ext/java/org/opentripplanner/ext/vectortiles/layers/areastops/AreaStopPropertyMapper.java

```diff
@@ -0,0 +1,52 @@
+package org.opentripplanner.ext.vectortiles.layers.areastops;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.opentripplanner.apis.support.mapping.PropertyMapper;
+import org.opentripplanner.framework.i18n.I18NStringMapper;
+import org.opentripplanner.inspector.vector.KeyValue;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.site.AreaStop;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.service.TransitService;
+
+public class AreaStopPropertyMapper extends PropertyMapper<AreaStop> {
+
+  private final Function<StopLocation, Collection<Route>> getRoutesForStop;
+  private final I18NStringMapper i18NStringMapper;
+
+  protected AreaStopPropertyMapper(
+    Function<StopLocation, Collection<Route>> getRoutesForStop,
+    Locale locale
+  ) {
+    this.getRoutesForStop = getRoutesForStop;
+    this.i18NStringMapper = new I18NStringMapper(locale);
+  }
+
+  protected static AreaStopPropertyMapper create(TransitService transitService, Locale locale) {
+    return new AreaStopPropertyMapper(transitService::getRoutesForStop, locale);
+  }
+
+  @Override
+  protected Collection<KeyValue> map(AreaStop stop) {
+    var routeColors = getRoutesForStop
+      .apply(stop)
+      .stream()
+      .map(Route::getColor)
+      .filter(Objects::nonNull)
+      .distinct()
+      // the MVT spec explicitly doesn't cover how to encode arrays
+      // https://docs.mapbox.com/data/tilesets/guides/vector-tiles-standards/
+      .collect(Collectors.joining(","));
+    return List.of(
+      new KeyValue("gtfsId", stop.getId().toString()),
+      new KeyValue("name", i18NStringMapper.mapNonnullToApi(stop.getName())),
+      new KeyValue("code", stop.getCode()),
```

Comment:
Stop code is optional, right? What also confuses me is that we have sometimes(?) returned "null" instead of null. Was there some issue with null?

Reply:
The underlying library code removes null values before serialising, so I'm not sure where the "null" string comes from. But yes, code is optional.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/3855

Reply:
So seems like this concern is probably not relevant here.

Reply:
I would like to get away from encoding complex objects at all. You should just use the regular APIs for that.